meerk40t/kernel/settings.py

```diff
@@ -19,11 +21,13 @@ class Settings:
     `write_configuration` is called.
     """
 
-    def __init__(self, directory, filename, ignore_settings=False):
+    def __init__(self, directory, filename, ignore_settings=False, create_backup=0):
         self._config_file = Path(get_safe_path(directory, create=True)).joinpath(
             filename
         )
         self._config_dict = {}
+        # 0 = No backup, 1 = keep daily, 2 = keep hourly, 3 = keep every minute
```

Comment:
Due to the way python stores strings, namely it creates a big table of them if they are explicit like "daily", "hourly", "minute" are basically equal to numbers. Since they will check `==` based on whether the objects are actually the same object if it's there. So really rather than numbers like this you can just use strings. (Though fstring versions of the same values only work at runtime, and so comparing fstrings is a longer algorithm.

Also, these are weird time periods.  And we should probably just make a copy of the last 2 or so previous settings files. Like you would only really want this if you messed them up, and then you'd have the previous ones.

Reply:
Done